datreant/collections.py

```diff
@@ -152,6 +167,48 @@ def names(self):
         return names
 
     @property
+    def basedirs(self):
+        """Return a list of member basedirs.
+
+        Members that can't be found will have basedir ``None``.
+
+        :Returns:
+            *names*
+                list giving the basedir of each member, in order;
+                members that are missing will have basedir ``None``
+
+        """
```

Comment:
what should also work is

```python
basedirs = [member.basedir if member else None for member in self._list()]
```

This comprehension did something similar in a short test
```python
[i if i %2 ==0 else None for i in range(8)]
```